rxandroid/src/main/java/io/reactivex/android/schedulers/AndroidSchedulers.java

```diff
@@ -47,6 +48,28 @@ public static Scheduler from(Looper looper) {
         return new HandlerScheduler(new Handler(looper));
     }
 
+    @TargetApi(Build.VERSION_CODES.P)
+    private static Handler createAsyncHandler(Looper looper) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            return Handler.createAsync(looper);
+        } else {
+            try {
+                //noinspection JavaReflectionMemberAccess
+                return Handler.class
+                    .getConstructor(Looper.class, Handler.Callback.class, boolean.class)
+                    .newInstance(looper, null, true);
+            } catch (IllegalAccessException e) {
+                return new Handler(looper);
```

Comment:
Just make these all empty and hoist a single `return` outside.

Reply:
Good catch 551b7b4